### DISCUSSION TITLE
TextBox StringFormat

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#11670] [Q&A] [answered]
Is it possible to use StringFormat on a TextBox binding and still have the TextBox be editable?
I'm trying to show a formatted number when the TextBox is unfocused, but show the unformatted number when editing the TextBox.

**Comment by timunie** [comment]
@SD-alexander-bittermann not out of the box I guess but there are some ideas:
1. Use a converter with convert back implemented
2. make use of `InnerLeftContent` and `InnerRightContent` property for things like units
3. use Avalonia.Behaviors to solve the issue

**Reply by ghost** [reply]
Avalonia.Behaviors seems like my only option since I want to show a different value depending on the focused state of the TextBox.
Specifically, when it's unfocused I want to show a formatted number (e.g. 9,999) and once I click the TextBox I want to show the unformatted number (e.g. 9999) and be able to edit without the commas.

I  tried something like this but haven't got it to work yet:
```
  <TextBox Name="NumberBox" TextAlignment="Right"  MaxLength="4"
                      Text="{Binding Number, StringFormat={}{0:n0}}">
      <Interaction.Behaviors>
          <EventTriggerBehavior EventName="LostFocus">
              <ChangeAvaloniaPropertyAction TargetProperty="{Binding #NumberBox.Text}"
                                            Value="{Binding Number, StringFormat={}{0:n0}}" />
          </EventTriggerBehavior>
          <EventTriggerBehavior EventName="GotFocus">
              <ChangeAvaloniaPropertyAction TargetProperty="{Binding #NumberBox.Text}"
                                            Value="{Binding Number}" />
          </EventTriggerBehavior>
      </Interaction.Behaviors>
  </TextBox>
```

Do you have any ideas how to make this work?
Thanks in advance!

**Reply by timunie** [reply, marked as answer]
Inside the convert back method you can also parse formatted text if you know format in advance. 

For behaviors, check out their samples provided in source code. These are really neat. Esp. The editable listbox sample is probably for you.

**Reply by ghost** [reply]
Thanks a lot, found a solution!

**Reply by timunie** [reply]
are you willing to share your solution for others having a similar usecase?
